py3status/modules/net_iplist.py

```diff
@@ -0,0 +1,152 @@
+        (default ['lo'])
+    iface_sep: string to write between interfaces.
+        (default ' ')
+    ip_blacklist: list of IPs to ignore. Accepts shell-style wildcards.
+        (default [])
+    ip_sep: string to write between IP addresses.
+        (default ',')
+    no_connection: string to show if there are no IPs to display.
+        (default 'no connection')
+
+Format placeholders:
+    {format_iface} the format_iface string.
+
+Format placeholders for format_iface:
+    {iface} name of the interface.
+    {ip} list of IPv4 of the interface.
+    {ip6} list of IPv6 of the interface.
+
+Color options:
+    color_bad: no IPs to show
+    color_good: IPs to show
+
+Requires:
+    iproute2
+"""
+
+
+import re
+from subprocess import check_output
+from fnmatch import fnmatch
+
+
+class Py3status:
+    cache_timeout = 30
+    format = 'Network: {format_iface}'
+    format_iface = '{iface}: v4{{{ip}}} v6{{{ip6}}}'
+    iface_blacklist = ['lo']
+    iface_sep = ' '
+    ip_blacklist = []
+    ip_sep = ','
+    no_connection = 'no connection'
+
+    def ip_list(self):
+        response = {
+            'cached_until': self.py3.time_in(seconds=self.cache_timeout),
+            'full_text': ''
+        }
+
+        connection = False
+        data = self._get_data()
+        iface_txt = ''
+        for iface, ips in data.items():
+            if iface in self.iface_blacklist:
+                continue
+
+            txt_ip = ''
+            txt_ip6 = ''
+            for ip in ips.get('ip', []):
+                if self._check_blacklist(ip):
+                    connection = True
+                    txt_ip += ip + self.ip_sep
+            txt_ip = txt_ip[:-len(self.ip_sep)]
+            for ip in ips.get('ip6', []):
+                if self._check_blacklist(ip):
+                    connection = True
+                    txt_ip6 += ip + self.ip_sep
+            txt_ip6 = txt_ip6[:-len(self.ip_sep)]
+            iface_txt += self.py3.safe_format(self.format_iface, {'iface': iface,
+                                                                  'ip': txt_ip,
+                                                                  'ip6': txt_ip6})
+            iface_txt += self.iface_sep
+            iface_txt = iface_txt[:-len(self.iface_sep)]
+        if not connection:
+            response['full_text'] = self.py3.safe_format(self.no_connection,
+                                                         {'format_iface': iface_txt})
+            response['color'] = self.py3.COLOR_BAD
+        else:
+            response['full_text'] = self.py3.safe_format(self.format,
+                                                         {'format_iface': iface_txt})
+            response['color'] = self.py3.COLOR_GOOD
+
+        return response
+
+    def _get_data(self):
+        iface_re = re.compile(r'\d+: (\w+):')
+        ip_re = re.compile(r'\s+inet (\d{1,3}.\d{1,3}.\d{1,3}.\d{1,3})/')
+        ip6_re = re.compile(r'\s+inet6 ([\da-f:]+)/')
```

Comment:
Probably not a huge performance deal but if you compiled these in `__init__()` then you would only need to do it once.

Did you consider using groups?  eg `(?P<name>....)`